dockerfiles/Dockerfile.build

```diff
@@ -0,0 +1,14 @@
+FROM openjdk:11
```

Comment:
Any reason for the Dockerfile not being in the root of the project?

Reply:
This is mostly because we're likely to end up with several so it seemed like a good idea to not leave them in the root. Also, it makes the dockerfile linting step in the Jenkinsfile a bit simpler - we can just tell it to lint everything in `/dockerfiles` rather than hard-coding the names of each dockerfile into the pipeline as (I think) we'd have to do if we just left them all in the root